test/test_distributions.py

```diff
@@ -1542,13 +1572,29 @@ def setUp(self):
 
     def test_kl_monte_carlo(self):
         set_rng_seed(0)  # see Note [Randomized statistical tests]
-        for p, q in self.finite_examples:
-            x = p.sample(sample_shape=(23000,))
+        for (p, _), (_, q) in self.finite_examples:
+            x = p.sample(sample_shape=(1000,))
             expected = (p.log_prob(x) - q.log_prob(x)).mean(0)
             actual = kl_divergence(p, q)
-            message = 'Incorrect KL({}, {}). expected {}, actual {}'.format(
-                type(p).__name__, type(q).__name__, expected, actual)
-            self.assertEqual(expected, actual, prec=0.1, message=message)
+            message = 'Incorrect KL({}, {}).\nExpected (Monte Carlo): {}\nActual (analytic): {}'.format(
+                    type(p).__name__, type(q).__name__, expected, actual)
+            try:
+                self.assertEqual(expected, actual, prec=0.1, message=message)
```

Comment:
Consider moving this to the `setUp(self)` above as `self.precision = 0.1` and adding a comment like
> These tests should pass with precision = 0.01, but that makes tests very expensive. Instead we test with precision = 0.1 and only test with higher precision when adding a new KL implementation.

Reply:
Okay.